server/auth_server.go

```diff
@@ -365,6 +552,79 @@ func UpgradeSettings() error {
 	return nil
 }
 
+func UpgradeCase() error {
+	var settings []string
+
+	config := model.AuthConfig{Resource: client.Resource{
+		Type: "config",
+	}}
+
+	//add the common settings
+	settings = append(settings, accessModeSetting)
+	settings = append(settings, allowedIdentitiesSetting)
+	settings = append(settings, securitySetting)
+	settings = append(settings, providerSetting)
+	settings = append(settings, providerNameSetting)
+
+	dbSettings, err := readCommonSettings(settings)
+	if err != nil {
+		log.Errorf("GetConfig: Error reading DB settings %v", err)
+		return err
+	}
+
+	config.AccessMode = dbSettings[accessModeSetting]
+	enabled, err := strconv.ParseBool(dbSettings[securitySetting])
+	if err == nil {
+		config.Enabled = enabled
+	} else {
+		config.Enabled = false
+	}
+
+	if enabled {
```

Comment:
why only save the settings if auth is currently enabled? Could we do it even if it were disabled?